src/test/fee_est/fee_est.cpp

```diff
@@ -0,0 +1,331 @@
+      "mark": "line",
+    }
+  ]
+}
+vega.embed("#vis", spec);
+</script>
+)";
+
+// Maximum number of randomly sampled transactions to show on graph.
+const size_t SAMPLE_TXS = 50000;
+
+// Drop 1/20th of sampled transactions with highest feerates if they are
+// outliers that would change the scale of the graph.
+const int DROP_FRAC = 20;
+
+void WriteGraph(const std::string& filename,
+    const TxMap& txMap,
+    const std::vector<unsigned int>& targets,
+    CBlockPolicyEstimator& estimator)
+{
+    UniValue fees(UniValue::VARR);
+
+    int64_t maxFeeRate = 0;
+    const unsigned int maxTarget = estimator.getMaxTarget();
+    for (bool conservative : {false, true}) {
+        for (unsigned int target : targets) {
+            if (target > maxTarget) continue;
+            UniValue fee(UniValue::VOBJ);
+            fee.pushKV("mode", conservative ? "conservative" : "default");
+            fee.pushKV("blocks", int(target));
+            CAmount feeRate = estimator.estimateSmartFee(target, nullptr /* fee_calc */, conservative).GetFeePerK();
+            fee.pushKV("feeRate", feeRate);
+            fees.push_back(fee);
+            maxFeeRate = std::max(maxFeeRate, feeRate);
+        }
+    }
+
+    std::vector<const TxMap::value_type*> sample;
+    sample.reserve(SAMPLE_TXS);
+    int i = 0;
+    std::minstd_rand randint;
+    for (const auto& entry : txMap) {
+        if (entry.second.actualBlocks > int(maxTarget)) {
+            continue;
+        } else if (sample.size() < SAMPLE_TXS) {
+            sample.emplace_back(&entry);
+        } else {
+            size_t j = randint() % i;
+            if (j < SAMPLE_TXS) {
+                sample[j] = &entry;
+            }
+        }
+        ++i;
+    }
+
+    if (!sample.empty()) {
+        std::vector<CAmount> feeRates;
+        feeRates.reserve(sample.size());
+        for (const auto& entry : sample) {
+            feeRates.emplace_back(CFeeRate(entry->second.fee, entry->second.size).GetFeePerK());
+        }
+        auto nth = feeRates.begin() + sample.size() / DROP_FRAC;
+        std::nth_element(feeRates.begin(), nth, feeRates.end(), std::greater<int64_t>());
+        maxFeeRate = std::max(maxFeeRate, *nth);
+    }
+
+    UniValue txs(UniValue::VARR);
+    for (const auto& entry : sample) {
+        if (entry->second.actualBlocks > 0) {
+            CAmount feeRate = CFeeRate(entry->second.fee, entry->second.size).GetFeePerK();
+            if (feeRate <= maxFeeRate) {
+                UniValue tx(UniValue::VOBJ);
+                tx.pushKV("feeRate", feeRate);
+                tx.pushKV("blocks", entry->second.actualBlocks);
+                tx.pushKV("height", entry->second.height);
+                txs.push_back(std::move(tx));
+            }
+        }
+    }
+
+    std::ofstream file(filename);
+    file << strprintf(GRAPH_HTML, txs.write(), fees.write());
+}
+
+bool UpdateCross(TxMap::value_type& tx, const std::vector<unsigned int>& targets, CBlockPolicyEstimator& estimator)
```

Comment:
538c780d078ecf400eb7b509f9fbcd176cdd290c
```suggestion
bool UpdateCross(TxMap::value_type& tx, const std::vector<unsigned int>& targets, const CBlockPolicyEstimator& estimator)
```